### PR TITLE
HPCC-13854 Allow PROJECT(ROW,TRANSFORM(SKIP)) to be optimized

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

### CODE REVIEW

**Comment by ghalliday** [comment]
@jamienoss please review

File: ecl/hqlcpp/hqlhtcpp.cpp [hunk 1]
```diff
@@ -16904,7 +16904,7 @@ ABoundActivity * HqlCppTranslator::doBuildActivityCreateRow(BuildCtx & ctx, IHql
     IHqlExpression * self = selfCursor->querySelector();
 
     if (isDataset)
-        associateSkipReturnMarker(funcctx, queryBoolExpr(false), selfCursor);
+        associateSkipReturnMarker(funcctx, queryZero(), selfCursor);
```
Comment (ghalliday) [inline ecl/hqlcpp/hqlhtcpp.cpp]: This was previously wrong (but would have worked).  The return type of the function is integer, not boolean.

**Comment by jamienoss** [comment]
https://track.hpccsystems.com/browse/HPCC-13854

**Comment by jamienoss** [comment]
@ghalliday should this really wait for 6.0.0 (merged to master)?

File: ecl/hqlcpp/hqlhtcpp.cpp (hunk 1 above)
Comment (jamienoss) [inline ecl/hqlcpp/hqlhtcpp.cpp]: The return type of which function? It looks like you don't mean a C++ function.

**Comment by ghalliday** [comment]
@jamienoss it isn't critical, and we should be aiming to close 5.4 down now

File: ecl/hqlcpp/hqlhtcpp.cpp (hunk 1 above)
Comment (richardkchapman) [inline ecl/hqlcpp/hqlhtcpp.cpp]: The transform function that it is generating - which returns 0 (not false) if it is to skip.

**Pull request merged**